controllers/helpers.go

```diff
@@ -303,7 +306,7 @@ func GetSubnetUUID(ctx context.Context, client *prismclientv3.Client, peUUID str
 
 // GetImage returns an image. If no UUID is provided, returns the unique image with the name.
 // Returns an error if no image has the UUID, if no image has the name, or more than one image has the name.
-func GetImage(ctx context.Context, client *prismclientv3.Client, id infrav1.NutanixResourceIdentifier) (*prismclientv3.ImageIntentResponse, error) {
+func GetImage(ctx context.Context, client *prismclientv3.Client, id *infrav1.NutanixResourceIdentifier) (*prismclientv3.ImageIntentResponse, error) {
```

Comment:
What's the reason to change the param to pointer?

Reply:
Image is now a pointer because it's a pointer on the nutanixMachine object. It is an optional item a user can pass in.

Reply:
Yeah, but we can check that Image is nil outside the function and call the function with `*id`. IMHO it's not necessary to change the function

Reply:
thats true. it would be unexpected to have a pointer type in the API and then have functions that require you to deference it before passing them as input.